web/src/main/java/org/cbioportal/web/util/AlterationCountUtil.java

```diff
@@ -0,0 +1,109 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class AlterationCountUtil {
+    @Autowired
+    private AlterationCountService alterationCountService;
+    @Autowired
+    private GenePanelService genePanelService;
+    
+    public Pair<List<AlterationCountByGene>, Long> getAlterationCounts(
+        AlterationCountFilter alterationCountFilter,
+        AlterationCountType alterationCountType,
+        List<Integer> entrezGeneIds
+    ) {
+        // select all genes if no list of entrez gene ids provided
+        Select<Integer> geneSelect = (entrezGeneIds == null || entrezGeneIds.isEmpty()) ?
+            Select.all(): Select.byValues(entrezGeneIds);
+        List<MolecularProfileCaseIdentifier> molecularProfileCaseIdentifiers =
+            alterationCountFilter.getMolecularProfileCaseIdentifiers();
+        boolean includeFrequency = true;
+        boolean includeMissingAlterationsFromGenePanel = false;
+        AlterationFilter alterationFilter = alterationCountFilter.getAlterationFilter();
+
+         return alterationCountType.equals(AlterationCountType.SAMPLE) ?
+            alterationCountService.getSampleAlterationCounts(
+                molecularProfileCaseIdentifiers,
+                geneSelect,
+                includeFrequency,
+                includeMissingAlterationsFromGenePanel,
+                alterationFilter
+            ) :
+            alterationCountService.getPatientAlterationCounts(
+                molecularProfileCaseIdentifiers,
+                geneSelect,
+                includeFrequency,
+                includeMissingAlterationsFromGenePanel,
+                alterationFilter
+            );
+    }
+    
+    public Set<GenePanelToGene> getProfiledGenes(
+        AlterationCountFilter alterationCountFilter,
+        AlterationCountType alterationCountType,
+        List<Integer> entrezGeneIds
+    ) {
+        List<MolecularProfileCaseIdentifier> molecularProfileCaseIdentifiers =
+            alterationCountFilter.getMolecularProfileCaseIdentifiers();
+        Set<GenePanelToGene> profiledGenes;
+        List<GenePanelData> genePanelDataList = alterationCountType.equals(AlterationCountType.SAMPLE) ?
+            genePanelService.fetchGenePanelDataInMultipleMolecularProfiles(molecularProfileCaseIdentifiers) :
+            genePanelService.fetchGenePanelDataInMultipleMolecularProfilesByPatientIds(molecularProfileCaseIdentifiers);
+
+        if (genePanelDataList
+            .stream()
+            .anyMatch(
+                genePanelData -> genePanelData.getProfiled() && genePanelData.getGenePanelId() == null
+            )
+        ) {
+            // if there is at least one sample/patient with all genes profiles,
+            // then all genes are profiled for the current data set
+            profiledGenes = null;
```

Comment:
I don't get this logic. I would assume that a method called `getProfiledGenes` would return all the genes profiled by the molecular profiles listed in the alt count filter. This logic of _if there exists a gene panel for these molecular profiles that was profiled, then return `null`_ just makes no sense to me. What is `null` representing here?

Reply:
I agree that returning `null` a bit confusing. We should return something like `all genes are profiled`, but not sure how to do that with a method that returns a set. This logic is inline with the front end code: https://github.com/cBioPortal/cbioportal-frontend/blob/3a8e6c89e43c1bbec138b36ba6bb62a910f1db44/src/shared/lib/GenePanelUtils.ts#L78-L108. We have the same assumption there. No `genePanelId` means `all genes profiled`. We should try refactoring some gene panel related stuff, but I think it is out of this PR's scope.

Reply:
Ok, lets at least put this in the method's javadoc. I have to revive my gene panel PR eventually; when I do, I'll chat with you about what we can do to clean this up